src/AppInstallerRepositoryCore/Rest/RestSourceFactory.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "pch.h"
+#include "RestSourceFactory.h"
+#include "RestClient.h"
+#include "RestSource.h"
+
+using namespace std::string_literals;
+using namespace std::string_view_literals;
+
+namespace AppInstaller::Repository::Rest
+{
+    namespace
+    {
+        // The base class for data that comes from a rest based source.
+        struct RestSourceFactoryBase : public ISourceFactory
+        {
+            std::shared_ptr<ISource> Create(const SourceDetails& details, IProgressCallback&) override final
+            {
+                THROW_HR_IF(E_INVALIDARG, details.Type != RestSourceFactory::Type());
+
+                RestClient restClient = RestClient::RestClient(details.Arg);
+
+                // TODO: Change identifier if required.
+                return std::make_shared<RestSource>(details, details.Arg, std::move(restClient));
+            }
+
+            void Add(SourceDetails& details, IProgressCallback&) override final
+            {
+                if (details.Type.empty())
+                {
+                    // With more than one source implementation, we will probably need to probe first

```

Comment:
>// With more than one source implementation, we will probably need to probe first [](start = 20, length = 81)

nit: is this comment copy pasted? Does not seem very relevant to what the code is doing.